src/java/com/sarality/app/view/list/SearchListComponentAdapter.java

```diff
@@ -0,0 +1,61 @@
+package com.sarality.app.view.list;
+
+import android.content.Context;
+import android.widget.Filter;
+import android.widget.Filterable;
+
+import com.sarality.app.common.collect.Lists;
+import com.sarality.app.view.action.ComponentActionManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Custom adapter that can filter the list and create new list
+ *
+ * @author sunayna@dothat.in sunayna
+ */
+public class SearchListComponentAdapter<T> extends ListComponentAdapter<T> implements Filterable {
+
+  private final List<T> rowValueList;
+  private final ComponentComparator comparator;
+
+  public SearchListComponentAdapter(Context context, ListRowRenderer<T> rowRenderer, List<T> rowValueList,
+                                    ComponentActionManager componentManager, ComponentComparator comparator) {
+
+    super(context, rowRenderer, rowValueList, componentManager);
+    // Make a coopy of the list that will not get manipulated when the list data is changed due to search
+    List<T> newList = Lists.of();
+    newList.addAll(rowValueList);
+    this.rowValueList = newList;
+    this.comparator = comparator;
+  }
+
+  public Filter getFilter() {
+    return new Filter() {
+
+      @Override
```

Comment:
Might want to create this as an inner class rather than anonymous one, especially given that it is the JVM that decides how many times getFilter gets called. Best to create an instance in the Constructor and return that instance in the call.